internal/lock.go

```diff
@@ -5,29 +5,48 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+	"time"
 )
 
-// Common load config
-func (c *Config) CheckLock(filename string) bool {
-	p := filepath.Join(c.Root, filename)
-	if _, err := os.Stat(p); os.IsNotExist(err) {
+// Lock enum
+type Lock string
+
+const (
+	PROOF_LOCK Lock = "proof.lock"
+	EPOCH_LOCK Lock = "epoch.lock"
+)
+
+func (l *Lock) toString() string {
+	return string(*l)
+}
+
+// Check if lock exists
+func (c *Config) CheckLock(lock Lock) bool {
+	p := filepath.Join(c.Root, lock.toString())
+	if stat, err := os.Stat(p); os.IsNotExist(err) {
 		return false
+	} else {
+		if time.Since(stat.ModTime()).Minutes() > 30 {
+			if err = c.RemoveLock(lock); err != nil {
+				return true
+			}
+		}
 	}
```

Comment:
Remove lock files if it exists more than 30 mins, one generating for an epoch is around 20+ mins